File: ruby/v2.4/get_dimension_values.rb
#!/usr/bin/env ruby
# Encoding: utf-8
#
# Copyright:: Copyright 2013, Google Inc. All Rights Reserved.
#
# License:: Licensed under the Apache License, Version 2.0 (the "License");
#           you may not use this file except in compliance with the License.
#           You may obtain a copy of the License at
#
#           http://www.apache.org/licenses/LICENSE-2.0
#
#           Unless required by applicable law or agreed to in writing, software
#           distributed under the License is distributed on an "AS IS" BASIS,
#           WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
#           implied.
#           See the License for the specific language governing permissions and
#           limitations under the License.
#
# This example illustrates how to list all values for a dimension

require_relative 'dfareporting_utils'

def get_dimension_values(dfareporting, args)
  # Create the dimension to query
  dimension = {
    :dimensionName => 'dfa:advertiser',
    :startDate => '2013-01-01',
    :endDate => '2013-12-31'
  }

  # Get all dimension values
  request = dfareporting.dimension_values.query(
    :profileId => args[:profile_id]
  ).body(dimension)

  loop do
    result = request.execute()

    # Display results
    result.data.items.each do |dimension|
      puts 'Dimension with ID %d and value "%s" was found.' %
          [dimension.id, dimension.value]
    end

    token = result.next_page_token

    break unless token and result.data.items.any?
    request = result.next_page
  end
end

if __FILE__ == $0
  # Retrieve command line arguments
  args = DfaReportingUtils.get_arguments(ARGV, :profile_id)

  # Authenticate and initialize API service
  dfareporting = DfaReportingUtils.setup()

  get_dimension_values(dfareporting, args)
end

File: python/v3_2/get_dimension_values.py
#!/usr/bin/python
#
# Copyright 2015 Google Inc. All Rights Reserved.
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#      http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.

"""This example illustrates how to get all dimension values for a dimension."""

import argparse
import sys

import dfareporting_utils
from oauth2client import client

# Declare command-line flags.
argparser = argparse.ArgumentParser(add_help=False)
argparser.add_argument(
    'profile_id', type=int,
    help='The ID of the profile to get a report for')


def main(argv):
  # Retrieve command line arguments.
  flags = dfareporting_utils.get_arguments(argv, __doc__, parents=[argparser])

  # Authenticate and construct service.
  service = dfareporting_utils.setup(flags)

  profile_id = flags.profile_id

  try:
    # Create the dimension to query
    dimension = {
        'dimensionName': 'dfa:advertiser',
        'startDate': '2012-01-01',
        'endDate': '2013-12-31'
    }

    # Construct the request.
    request = service.dimensionValues().query(
        profileId=profile_id, body=dimension)

    while True:
      # Execute request and print response.
      response = request.execute()

      for value in response['items']:
        print ('Found dimension value with ID %s and value "%s".'
               % (value['id'], value['value']))

      if response['items'] and response['nextPageToken']:
        request = service.dimensionValues().query_next(request, response)
      else:
        break

  except client.AccessTokenRefreshError:
    print ('The credentials have been revoked or expired, please re-run the '
           'application to re-authorize')


if __name__ == '__main__':
  main(sys.argv)
